lib/components/dashboard/degree-info/DistributionBarsJSX.tsx

```diff
@@ -199,25 +199,29 @@ const DistributionBarsJSX: FC<{ major: Major }> = ({ major }) => {
     // If double_count is string[], the specified distributions / fine requirements are 'whitelisted'
     // When a course satisfies a distribution, double_count takes the value of distribution.double_count
     if (!courseObj) return;
-    let distDoubleCount: string[] | undefined = undefined; // initial value
+    let distDoubleCount: string[] | undefined = ['All']; // double count by defualt
     reqs.forEach((reqGroup, i) => {
       let req = reqGroup[1][0]; // general distribution
       if (
-        (!distDoubleCount || distDoubleCount.includes(req.name)) &&
+        distDoubleCount &&
+        (distDoubleCount.includes(req.name) ||
+          distDoubleCount.includes('All')) &&
         (req.fulfilled_credits < req.required_credits ||
           (req.required_credits === 0 && req.fulfilled_credits === 0)) &&
         checkRequirementSatisfied(req, courseObj)
       ) {
         reqs[i][1][0].fulfilled_credits += parseInt(courseObj.credits);
         distDoubleCount = req.double_count; // set double_count, if any
-        let fineDoubleCount: string[] | undefined = undefined;
+        let fineDoubleCount: string[] | undefined = ['All'];
         reqGroup[1].forEach((req: requirements, j: number) => {
           // fine reqs
```

Comment:
The fine reqs nesting is a bit hard to follow. Can you add comments explaining what's happening and also take finereqs out of the nesting (flatten) or take it out as a helper method (unflattened)?

Reply:
good idea, taken out as helper method